homeassistant/components/eheimdigital/entity.py

```diff
@@ -46,6 +47,14 @@ def __init__(
         self._device = device
         self._device_address = device.mac_address
 
+    @classmethod
+    def _get_model_name(cls, device: EheimDigitalDevice) -> str | None:
+        if isinstance(device, EheimDigitalFilter) and device.filter_model_name:
+            return device.filter_model_name
+        if device.device_type.model_name:
+            return device.device_type.model_name
+        return None
```

Comment:
Note, isinstance checks can be expensive. Would it make more sense in the future to have `EheimDigitalDevice` implement a method that just returns the model name and then `EheimDigitalFilter` can implement that differently?